src/seedu/addressbook/Main.java

```diff
@@ -112,11 +112,18 @@ private CommandResult executeCommand(Command command)  {
             storage.save(addressBook);
             return result;
         } catch (Exception e) {
+            if(checkIfFileIsReadOnly(e)){
```

Comment:
No need to check for the writability explicitly. Exception already gives you that information. 
All you need to do is to handle it!